### PR TITLE
(FACT-2892) Add extra search for primary interface

### PR DESCRIPTION
**Pull request by sebastian-miclea**

### CODE REVIEW

**Comment by sebastian-miclea** [comment]
Jenkins please test this on all

**Comment by sebastian-miclea** [comment]
Jenkins please test this on all

File: lib/facter/resolvers/solaris/networking.rb [hunk 1]
```diff
@@ -25,9 +24,14 @@ def read_facts(fact_name)
                 obtain_info_for_interface(lifreq)
               end
 
-              @fact_list = { interfaces: @interfaces } unless @interfaces.empty?
               @fact_list[:primary_interface] = Facter::Util::Resolvers::Networking::PrimaryInterface.read_from_route
 
+              unless @interfaces.empty?
+                @fact_list = { interfaces: @interfaces }
+                @fact_list[:primary_interface] ||=
+                  Facter::Util::Resolvers::Networking::PrimaryInterface.find_in_interfaces(@interfaces)
```
Comment (oanatmaria) [inline lib/facter/resolvers/solaris/networking.rb:32]: Something similiar should be implemented for AIX networking_resolver

Comment (sebastian-miclea) [reply lib/facter/resolvers/solaris/networking.rb:32]: AIX uses one call to get both primary and all bindings, if that fails, interfaces won't be available to search for a primary.

**Pull request merged**